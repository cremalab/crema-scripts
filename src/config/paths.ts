import path from "path"

const hygen = path.join(__dirname, "../..", "node_modules/.bin/hygen")
const reactScripts = path.join(
  __dirname,
  "../..",
  "node_modules/.bin/react-scripts",
)
const appRoot = path.resolve(process.cwd())
const appTemplates = path.join(appRoot, "_templates")
const root = path.join(__dirname, "../..")
const templates = path.join(root, "/_templates")

export const paths = {
  appRoot,
  appTemplates,
  hygen,
  reactScripts,
  root,
  templates,
}
