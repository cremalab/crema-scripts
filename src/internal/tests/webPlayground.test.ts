import spawn from "cross-spawn"
import { pathPackage } from "../config/paths"
import path from "path"
import rimraf from "rimraf"

describe("structure web", () => {
  beforeAll(() => {
    rimraf.sync("temp")
    spawn.sync("mkdir", ["temp"])
    process.chdir("temp")

    spawn.sync("node", [
      path.join(pathPackage(), "dist/projectData/web/index.js"),
      "new",
      "project",
    ])
  })

  afterAll(() => {
    process.chdir("..")
    rimraf.sync("temp")
  })

  test("runs playground", async () => {
    await spawn.sync("node", [
      path.join(pathPackage(), "dist/projectData/web/index.js"),
      "playgroundBuild",
    ])
    const tree = spawn.sync("treee", ["-a", "-l 6"]).output.toString()
    return expect(tree).toMatchSnapshot()
  })
})
