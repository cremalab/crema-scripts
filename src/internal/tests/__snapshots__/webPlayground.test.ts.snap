// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`structure web runs playground 1`] = `
",
/Users/Rob_LaFeve/Dev/crema-toolbox/src/internal/tests/temp
â”œâ”€â”€ .eslintrc.js
â”œâ”€â”€ .github
|  â””â”€â”€ PULL_REQUEST_TEMPLATE.md
â”œâ”€â”€ .prettierrc.js
â”œâ”€â”€ public
|  â”œâ”€â”€ favicon.ico
|  â”œâ”€â”€ index.html
|  â””â”€â”€ manifest.json
â”œâ”€â”€ src
|  â”œâ”€â”€ assets
|  |  â””â”€â”€ crema-logo.svg
|  â”œâ”€â”€ components
|  |  â”œâ”€â”€ App
|  |  |  â”œâ”€â”€ README.md
|  |  |  â”œâ”€â”€ __snapshots__
|  |  |  |  â””â”€â”€ test.tsx.snap
|  |  |  â”œâ”€â”€ index.css
|  |  |  â”œâ”€â”€ index.tsx
|  |  |  â”œâ”€â”€ stories.tsx
|  |  |  â””â”€â”€ test.tsx
|  |  â””â”€â”€ Initializer
|  |     â”œâ”€â”€ README.md
|  |     â”œâ”€â”€ __snapshots__
|  |     |  â””â”€â”€ test.tsx.snap
|  |     â”œâ”€â”€ index.tsx
|  |     â””â”€â”€ test.tsx
|  â”œâ”€â”€ crema-toolbox-env.d.ts
|  â”œâ”€â”€ index.css
|  â”œâ”€â”€ index.tsx
|  â”œâ”€â”€ serviceWorker.ts
|  â”œâ”€â”€ state
|  |  â”œâ”€â”€ actions
|  |  |  â”œâ”€â”€ actionTodoAdd.ts
|  |  |  â”œâ”€â”€ actionTodoRemove.ts
|  |  |  â””â”€â”€ actionTodoUpdate.ts
|  |  â”œâ”€â”€ index.ts
|  |  â””â”€â”€ reducers
|  |     â”œâ”€â”€ index.ts
|  |     â””â”€â”€ reducerTodos
|  |        â”œâ”€â”€ __snapshots__
|  |        |  â””â”€â”€ test.ts.stnap
|  |        â”œâ”€â”€ index.ts
|  |        â””â”€â”€ test.ts
|  â””â”€â”€ types
|     â”œâ”€â”€ ActionType.ts
|     â”œâ”€â”€ EntityTodo.ts
|     â”œâ”€â”€ State.ts
|     â””â”€â”€ Todo.ts
â”œâ”€â”€ storybook-static
|  â”œâ”€â”€ favicon.ico
|  â”œâ”€â”€ index.html
|  â”œâ”€â”€ main.df983bacd37249bda20f.bundle.js
|  â”œâ”€â”€ runtime~main.17da544d0ef88cbabfaf.bundle.js
|  â”œâ”€â”€ sb_dll
|  |  â”œâ”€â”€ storybook_ui-manifest.json
|  |  â”œâ”€â”€ storybook_ui_dll.LICENCE
|  |  â””â”€â”€ storybook_ui_dll.js
|  â””â”€â”€ vendors~main.c5900a230f247ca62e95.bundle.js
â””â”€â”€ tsconfig.json

directory: 17 file: 42 

[2K[1G,"
`;
